src/test/java/com/mongodb/InterruptibilityItTest.java

```diff
@@ -0,0 +1,277 @@
+            com.mongodb.assertions.Assertions.assertTrue(cursor.available() == 1);
+            cursor.next();
+            com.mongodb.assertions.Assertions.assertTrue(cursor.available() == 0);
+            assertThrowInVirtualInterruptedThread(MongoInterruptedException.class,
+                    interruptDelay, "getMore", cursor::next);
+        }
+    }
+
+    @ParameterizedTest
+    @MethodSource("arguments")
+    void serverSelection(final Duration interruptDelay) {
+        // we assume here that the time to elect a new leader is greater than `interruptDelay`
+        client.getDatabase("admin").runCommand(BsonDocument.parse(
+                "{\n"
+                // using `blockTimeout` does not give us much because it controls only the unelectability of the current primary
+                + "    'replSetStepDown': " + blockTimeout.toSeconds() + ",\n"
+                + "    'force': true\n"
+                + "}"));
+        MongoIterable<String> iterable = client.listDatabaseNames();
+        assertThrowInVirtualInterruptedThread(MongoInterruptedException.class,
+                interruptDelay, null, iterable::first);
+    }
+
+    private static Stream<Arguments> arguments() {
+        return Stream.of(
+                Arguments.of(Duration.ZERO),
+                Arguments.of(InterruptibilityItTest.interruptDelay)
+        );
+    }
+
+    @SuppressWarnings({"try", "UnusedReturnValue"})
+    private static <T extends Throwable> T assertThrowInVirtualInterruptedThread(
+            final Class<T> expectedType,
+            final Duration interruptDelay,
+            @Nullable final String blockedCommandName,
+            final Runnable action) {
+        try (@SuppressWarnings("unused") FailPoint fp = blockedCommandName == null ? null : FailPoint.enable(BsonDocument.parse(
+                "{'configureFailPoint': 'failCommand',\n"
+                + "    'appName': '" + clientSettings.getApplicationName() + "',\n"
+                + "    'mode': {\n"
+                + "        'times': 1"
+                + "    },\n"
+                + "    'data': {\n"
+                + "        'failCommands': [\n"
+                + "            '" + blockedCommandName + "'\n"
+                + "        ],\n"
+                + "        'blockConnection': true,\n"
+                + "        'blockTimeMS': " + blockTimeout.toMillis() + "\n"
+                + "    }\n"
+                + "}"),
+                primary(client, testTimeout),
+                clientSettingsBuilderSupplier)) {
+            return assertThrows(expectedType, () ->
+                    inVirtualThread(() ->
+                            inInterruptedThread(interruptDelay, action))
+            );
+        }
+    }
+
+    private static void inVirtualThread(final Runnable action) throws TimeoutException {
+        try (ExecutorService ex = Executors.newVirtualThreadPerTaskExecutor()) {
```

Comment:
If you replace `newVirtualThreadPerTaskExecutor` with `newCachedThreadPool`, all but the `serverSelection` tests start failing.